### PR TITLE
[main] [cryptsetup] Update cryptsetup to 2.4.3

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Updates cryptsetup to the latest stable release (2.4.3).
**NOTE:** Disabled the new ssh token feature since we don't have libssh in 2.0 at this time.

###### Change Log  

- Update cryptsetup to 2.4.3.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology
Local build, waiting on pipeline build

### CODE REVIEW

File: SPECS/cryptsetup/cryptsetup.spec [hunk 1]
```diff
@@ -1,135 +1,146 @@
...
 %{_tmpfilesdir}/cryptsetup.conf
 %ghost %dir /run/cryptsetup
- 
+
 %changelog
-* Thu Dec 16 2021 Pawel Winogrodzki <pawelwi@microsoft.com> - 2.3.3-3
-- Removing the explicit %%clean stage.
+* Tue Jan 18 2022 Daniel McIlvaney <damcilva@microsoft.com> - 2.4.3-1
+- Update to version 2.4.3
+- We currently do not support SSH tokens, re-enable once libssh is available
```
Comment (mfrw) [inline SPECS/cryptsetup/cryptsetup.spec:143]: Should we create a bug to track it if not already created.

Comment (dmcilvaney) [reply SPECS/cryptsetup/cryptsetup.spec:143]: Not a bad idea, I'll touch base with Jon and do so.

Comment (dmcilvaney) [reply SPECS/cryptsetup/cryptsetup.spec:143]: Done

**Comment by dmcilvaney** [comment]
I intend to ignore the linter suggestion for moving the %global, it conflicts with the more recent versions of the linter which will want it placed back at the top.

File: SPECS/cryptsetup/cryptsetup.spec [hunk 2]
```diff
@@ -1,135 +1,146 @@
...
 %find_lang cryptsetup
- 
+
 %post -n cryptsetup-libs -p /sbin/ldconfig
- 
 %postun -n cryptsetup-libs -p /sbin/ldconfig
- 
+
 %files
 %{!?_licensedir:%global license %%doc}
```
Comment (thcrain-msft) [inline SPECS/cryptsetup/cryptsetup.spec:102]: Nit: macro definition will never be needed, since we set the license directory in all versions of Mariner.

**Review by thcrain-msft — APPROVED** [review]

File: SPECS/cryptsetup/cryptsetup.spec (hunk 2 above)
Comment (dmcilvaney) [reply SPECS/cryptsetup/cryptsetup.spec:102]: Thanks, I'll clean that up.

**Pull request merged**